app/features/NotificationSlice.ts

```diff
@@ -0,0 +1,85 @@
+import { createSlice, Dispatch, PayloadAction } from '@reduxjs/toolkit';
+
+export enum NotificationLevel {
+    Info,
+    Error,
+    Update,
+}
+
+let nextId = 0;
+
+interface NotificationBase {
+    id: number;
+}
+
+interface UpdateNotification extends NotificationBase {
+    level: NotificationLevel.Update;
+}
+
+export interface Notification extends NotificationBase {
+    level: Exclude<NotificationLevel, NotificationLevel.Update>;
+    message: string;
+}
+
+interface NotificationSliceState {
+    notifications: (UpdateNotification | Notification)[];
+}
+
+const { actions, reducer } = createSlice({
+    name: 'notification',
+    initialState: {
+        notifications: [],
+    } as NotificationSliceState,
+    reducers: {
+        pushNotification(state, action: PayloadAction<Notification>) {
+            state.notifications.push({ ...action.payload });
+        },
+        triggerUpdateNotification(state) {
+            state.notifications.push({
+                level: NotificationLevel.Update,
+                id: nextId,
+            });
+
+            nextId += 1;
+        },
+        removeNotification(state, action: PayloadAction<number>) {
+            state.notifications = state.notifications.filter(
+                (n) => n.id !== action.payload
+            );
+        },
+    },
+});
+
+export default reducer;
+
+export const { triggerUpdateNotification, removeNotification } = actions;
+
+/**
+ * Display notifications of different types.
+ *
+ * @example
+ * const dispatch = useDispatch();
+ *
+ * pushNotification(dispatch, { NotificationLevel.Info, message: 'Informative message' }, 5);
+ * pushNotification(dispatch, { NotificationLevel.Update });
```

Comment:
You can't add an update notification with pushNotification, because of the types.
Either remove this from the example or update the types to allow it.

Reply:
True, I removed it as the update notification is meant to be triggered through the exported action instead.